### PR TITLE
Add multiple templates management on docList annotation

### PR DESCRIPTION
**Pull request by samakhov**
Tests are based on memory FileSystem

### CODE REVIEW

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/createTemplateFile.st [hunk 1]
```diff
@@ -0,0 +1,7 @@
+private
+createTemplateFile
+	| dir templateFile |
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/createTemplateFile.st]: I do not understand. We crate the file if it does not exist? Why not doing everything in memory directly?

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithEmptyMultipleMustacheTemplateShouldNotContainsRaw.st [hunk 2]
```diff
@@ -0,0 +1,18 @@
+private
+testAnnotationWithEmptyMultipleMustacheTemplateShouldNotContainsRaw
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithEmptyMultipleMustacheTemplateShouldNotContainsRaw.st]: I'd add a test comment showing the annotation we are creating. Something like:

"${documentList:....}"

Like that we understand what is expected more clrearly and we don't need to mentally parse the creation of the annotation.

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithLowLimitShouldOnlyTakeLimitNumberOfFilesInPath.st [hunk 3]
```diff
@@ -0,0 +1,24 @@
+private
+testAnnotationWithLowLimitShouldOnlyTakeLimitNumberOfFilesInPath
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithLowLimitShouldOnlyTakeLimitNumberOfFilesInPath.st]: Same as above

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithTooHighLimitShouldOnlyTakeNumberOfFilesInPath.st [hunk 4]
```diff
@@ -0,0 +1,25 @@
+private
+testAnnotationWithTooHighLimitShouldOnlyTakeNumberOfFilesInPath
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithTooHighLimitShouldOnlyTakeNumberOfFilesInPath.st]: Same as above

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithWrongTemplateExtensionShouldNotContainsRaw.st [hunk 5]
```diff
@@ -0,0 +1,21 @@
+private
+testAnnotationWithWrongTemplateExtensionShouldNotContainsRaw
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithWrongTemplateExtensionShouldNotContainsRaw.st]: Same as above

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithoutLimitParameterShouldTakeDefaultLimitValue.st [hunk 6]
```diff
@@ -1,6 +1,6 @@
 private
-testDocumentListWithOnlySortParameter
-	| doc documentGroup sections |
+testAnnotationWithoutLimitParameterShouldTakeDefaultLimitValue
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithoutLimitParameterShouldTakeDefaultLimitValue.st]: Same as above

File: src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithoutSortParameterShouldListFilesInArbitraryOrder.st [hunk 7]
```diff
@@ -1,5 +1,5 @@
 private
-testDirectoryDoesnotContainsEnoughFilesShouldWork
+testAnnotationWithoutSortParameterShouldListFilesInArbitraryOrder
```
Comment (guillep) [inline src/Pillar-Tests-ExporterCore.package/PRDocumentListTransformerTest.class/instance/testAnnotationWithoutSortParameterShouldListFilesInArbitraryOrder.st]: Same as above

**Comment by samakhov** [comment]
I added some comments in tests.
For createTemplateFile, It is for testing when the user specify a template file and it exists. The testing file system used in that method is a MemoryFileSystem. We are not creating a mustache template during execution. If the specified ones does not exist we only extract elements without a specific template.
For createDocList, I tried to refactor :-)

**Comment by samakhov** [comment]
I'm guille

**Pull request merged**